magento/magento2
# magento/magento2#33422: Fixed excel writer produces invalid files for numbers preceded by spaces

### Description (*)

When creating Excel, the is_numeric function is used to determine the cell type. The is_numeric function returns true for strings of the type "123", "\t123" etc. Then the cells are given the Number type, which when viewed leads to an error and the data is displayed as NaN.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#33422

### Manual testing scenarios (*)

1. Create an instance of Magento\Framework\Convert\Excel with an ArrayIterator over an array containing a field with a value like ␣123 (i.e. space followed by digits, read ␣ as space).
2. Call convert on the object.
3. Try opening the resulting file with a current version of Microsoft Excel or LibreOffice Calc. Previously the value was NaN, now the column type is changed to String and the value is displayed correctly.

### Questions or comments

Although the is_numeric and ctype_space functions change their behavior in 8.1, these changes should work correctly on both 7.4 and 8.1.

### Contribution checklist (*)
